src/openrct2-ui/windows/EditorInventionsList.cpp

```diff
@@ -94,27 +95,34 @@ static void ResearchRidesSetup()
     }
 }
 
-static std::pair<StringId, Formatter> WindowEditorInventionsListPrepareName(const ResearchItem& researchItem, bool withGap)
+static void DrawResearchItem(
+    rct_drawpixelinfo& dpi, const ResearchItem& researchItem, const int16_t& width, const ScreenCoordsXY& screenCoords,
+    StringId format, TextPaint textPaint)
 {
-    StringId drawString;
-    StringId stringId = researchItem.GetName();
+    const StringId itemNameId = researchItem.GetName();
+    int16_t columnSplitOffset = width / 2;
     auto ft = Formatter();
 
     if (researchItem.type == Research::EntryType::Ride
         && !GetRideTypeDescriptor(researchItem.baseRideType).HasFlag(RIDE_TYPE_FLAG_LIST_VEHICLES_SEPARATELY))
     {
-        drawString = withGap ? STR_INVENTIONS_LIST_RIDE_AND_VEHICLE_NAME_DRAG : STR_WINDOW_COLOUR_2_STRINGID_STRINGID;
-        StringId rideTypeName = get_ride_naming(researchItem.baseRideType, get_ride_entry(researchItem.entryIndex)).Name;
+        const StringId rideTypeName = get_ride_naming(researchItem.baseRideType, get_ride_entry(researchItem.entryIndex)).Name;
+
+        // Draw group name
         ft.Add<StringId>(rideTypeName);
-        ft.Add<StringId>(stringId);
+        DrawTextEllipsised(&dpi, screenCoords, columnSplitOffset - 11, format, ft, textPaint);
+
+        // Draw vehicle name
+        ft.Add<StringId>(itemNameId);
```

Comment:
Are you sure the `ft` Formatter doesn't need to be reset here like it used to? It probably does.

Reply:
Oh it's very true. Vehicle name was just being appended there. Good catch.